add warning note for downgrading on automated upgrade (#167)
k3s-io/docs: docs/upgrades/automated.md

kubectl -n system-upgrade get jobs -o yaml
```

## Downgrade Prevention

> :::info Version Gate
> Starting with the 2023-07 releases ([v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1))
> :::
> Kubernetes does not support downgrades of control-plane components. The k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.

Reviewer: I'm not sure if it's just GH not handling these blocks properly or if they will actually be rendered like this to the site, but I think you need newlines before and after this block.
```suggestion

:::info Version Gate
Starting with the 2023-07 releases ([v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1))

:::

Kubernetes does not support downgrades of control-plane components. The k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.
```

kubectl -n system-upgrade get jobs -o yaml
```

:::Warning

> Kubernetes does not support downgrades of control-plane components. Starting with [v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1) the k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.

Reviewer: final nit: this is the format we've used in other places, such as https://github.com/k3s-io/docs/blob/782949f994b0b21002b36dfbf4156b84607ed18a/docs/installation/network-options.md?plain=1#L35
```suggestion
Kubernetes does not support downgrades of control-plane components. Starting with the 2023-07 releases ([v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1)) the k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.
```

kube-system      traefik-64f55bb67d-4mm6s                                          1/1     Running   0          19h
system-upgrade   apply-k3s-server-on-ip-172-31-0-16-with-7af95590a5af8e8c3-2cdc6   0/1     Error     0          9m25s
system-upgrade   apply-k3s-server-on-ip-172-31-10-23-with-7af95590a5af8e8c-9xvwg   0/1     Error     0          14m
system-upgrade   apply-k3s-server-on-ip-172-31-13-213-with-7af95590a5af8e8-8j72v   0/1     Error     0          18m
system-upgrade   system-upgrade-controller-7c4b84d5d9-kkzr6                        1/1     Running   0          20m
```
and your nodes something like this:
```
NAME               STATUS                     ROLES                       AGE   VERSION
ip-172-31-0-16     Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-10-23    Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-13-213   Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-2-13     Ready                      <none>                      19h   v1.27.4+k3s1
```

> You can make your node schedulable again by simpling uncordoning target node with the command `kubectl uncordon NODE_NAME` or you can set the version on your plan file to a newer one so k3s-upgrade can attempt again to upgrade your version.

Reviewer: What happens if they uncordon the node but don't fix the plan version or channel? Will it cordon the node again next time it tries?
```suggestion
You can return your node to service by uncordoning it with the command `kubectl uncordon NODE_NAME`, or by changing the version or channel on your plan to target a release that is the same or newer than what is currently running on the cluster.
```

Author: it will cordon again the next try, i'll upgrade it to only show changing the version or channel

Reviewer: Maybe suggest:
* Abort the invalid upgrade by deleting the plan and manually uncordoning the nodes.
* Correct the upgrade by changing the version or channel in the pan.

kubectl -n system-upgrade get jobs -o yaml
```

> :::Warning
> Kubernetes does not support downgrades of control-plane components. Starting with the 2023-07 releases ([v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1)) the k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.
> 
> If you attempted it your pods in cluster should look something like this:
> ```

Reviewer 2: ```suggestion
## Downgrade Prevention
:::info Version Gate
Starting with the 2023-07 releases ([v1.27.4+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.27.4%2Bk3s1), [v1.26.7+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.26.7%2Bk3s1), [v1.25.12+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.25.12%2Bk3s1), [v1.24.16+k3s1](https://github.com/k3s-io/k3s-upgrade/releases/tag/v1.24.16%2Bk3s1))
:::

Kubernetes does not support downgrades of control-plane components. The k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.

If you attempted it your pods in cluster should look something like this:
```
```

Kubernetes does not support downgrades of control-plane components. The k3s-upgrade image used by upgrade plans will refuse to downgrade K3s, failing the plan and leaving your nodes cordoned.

> If you attempted it your pods in cluster should look something like this:

Reviewer: ```suggestion
Here is an example cluster, showing failed upgrade pods and cordoned nodes:
```

> ```
> ubuntu@user:~$ kubectl get pods -A
> NAMESPACE        NAME                                                              READY   STATUS    RESTARTS   AGE
> kube-system      coredns-77ccd57875-9ng74                                          1/1     Running   0          19h
> kube-system      local-path-provisioner-957fdf8bc-9vwzn                            1/1     Running   0          19h
> kube-system      metrics-server-648b5df564-wzbnh                                   1/1     Running   0          19h
> kube-system      svclb-traefik-0bda8e84-hbjq8                                      2/2     Running   0          19h
> kube-system      svclb-traefik-0bda8e84-jg94l                                      2/2     Running   0          19h
> kube-system      svclb-traefik-0bda8e84-qkcs7                                      2/2     Running   0          19h
> kube-system      svclb-traefik-0bda8e84-tfhjq                                      2/2     Running   0          19h
> kube-system      traefik-64f55bb67d-4mm6s                                          1/1     Running   0          19h
> system-upgrade   apply-k3s-server-on-ip-172-31-0-16-with-7af95590a5af8e8c3-2cdc6   0/1     Error     0          9m25s
> system-upgrade   apply-k3s-server-on-ip-172-31-10-23-with-7af95590a5af8e8c-9xvwg   0/1     Error     0          14m
> system-upgrade   apply-k3s-server-on-ip-172-31-13-213-with-7af95590a5af8e8-8j72v   0/1     Error     0          18m
> system-upgrade   system-upgrade-controller-7c4b84d5d9-kkzr6                        1/1     Running   0          20m
> ```
> and your nodes something like this:
> ```
> NAME               STATUS                     ROLES                       AGE   VERSION
> ip-172-31-0-16     Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
> ip-172-31-10-23    Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
> ip-172-31-13-213   Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
> ip-172-31-2-13     Ready                      <none>                      19h   v1.27.4+k3s1
> ```

Reviewer: Just show the system-upgrade pods
```suggestion
```console
ubuntu@user:~$ kubectl get pods -n system-upgrade
NAME                                                              READY   STATUS    RESTARTS   AGE
apply-k3s-server-on-ip-172-31-0-16-with-7af95590a5af8e8c3-2cdc6   0/1     Error     0          9m25s
apply-k3s-server-on-ip-172-31-10-23-with-7af95590a5af8e8c-9xvwg   0/1     Error     0          14m
apply-k3s-server-on-ip-172-31-13-213-with-7af95590a5af8e8-8j72v   0/1     Error     0          18m
system-upgrade-controller-7c4b84d5d9-kkzr6                        1/1     Running   0          20m

ubuntu@user:~$ kubectl get nodes
NAME               STATUS                     ROLES                       AGE   VERSION
ip-172-31-0-16     Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-10-23    Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-13-213   Ready,SchedulingDisabled   control-plane,etcd,master   19h   v1.27.4+k3s1
ip-172-31-2-13     Ready                      <none>                      19h   v1.27.4+k3s1
```
```

> You can return your node to service by deleting the plan `kubectl get plans -n system-upgrade` to get the plan name then `kubectl delete plans PLAN_NAME -n system-upgrade` then uncordoning it with the command `kubectl uncordon NODE_NAME`, or by changing the version or channel on your plan to target a release that is the same or newer than what is currently running on the cluster.

Reviewer: ```suggestion
You can return your cordoned nodes to service by either of the following methods:
 * Change the version or channel on your plan to target a release that is the same or newer than what is currently running on the cluster, so that the plan succeeds.
* Delete the plan and manually uncordon the nodes.
    Use `kubectl get plan -n system-upgrade` to find the plan name, then `kubectl delete plan -n system-upgrade PLAN_NAME` to delete it. Once the plan has been deleted, use `kubectl uncordon NODE_NAME` to uncordon each of the nodes.
```

kubectl -n system-upgrade get jobs -o yaml
```

## Warning

> Downgrading your cluster was never supported and since versions below k3s-upgrade will cordon your nodes when downgrade is attempted:

Reviewer 2: Please use the admonitions style we have for all other version gate notices. https://github.com/k3s-io/docs/blob/main/docs/cli/secrets-encrypt.md?plain=1#L11-L13
